server/document.js

```diff
@@ -162,9 +162,42 @@ export class NextScript extends Component {
     return [...this.getChunkScript('main.js', { async: true })]
   }
 
+  static getInlineScriptSource (documentProps) {
+    let { __NEXT_DATA__, chunks } = documentProps
+    __NEXT_DATA__ = Object.assign({
+      chunks: chunks.names
+    }, __NEXT_DATA__);
+    const { page, pathname } = __NEXT_DATA__
+
+    return `
+      __NEXT_DATA__ = ${htmlescape(__NEXT_DATA__)}
+      module={}
+      __NEXT_LOADED_PAGES__ = []
+      __NEXT_LOADED_CHUNKS__ = []
+
+      __NEXT_REGISTER_PAGE = function (route, fn) {
+        __NEXT_LOADED_PAGES__.push({ route: route, fn: fn })
+      }
+
+      __NEXT_REGISTER_CHUNK = function (chunkName, fn) {
```

Comment:
For some reason, you're adding back obsolete code that was removed in latest canary, maybe you created a branch based on master.

Here's the correct code: https://github.com/zeit/next.js/blob/canary/server/document.js#L186-L200

Reply:
Yes, I was not aware that development is not happening on master, I will move to the canary branch.